apply_recommendations/Dockerfile

```diff
@@ -1,7 +1,14 @@
-FROM hashicorp/terraform:1.7
+FROM golang:1.22.5-bookworm AS build
 
-RUN apk add --no-cache jq
+COPY . /opt
+WORKDIR /opt
 
-COPY --chmod=755 apply_recommendations.sh /opt/apply_recommendations.sh
+RUN CGO_ENABLED=0 GOOS=linux go build -o /apply-recommendations cmd/apply/main.go
 
-ENTRYPOINT ["/opt/apply_recommendations.sh"]
+FROM gcr.io/distroless/base-debian11
```

Comment:
sorry, I'm not familiar with what this image contains, isn't "distroless" and "debian" kind of a contradiction?

Reply:
Good point 😆 

I pulled it from [here](https://github.com/GoogleContainerTools/distroless?tab=readme-ov-file#docker); it sounded like about the right image to use.